### PR TITLE
Mintlify migration script

### PR DESCRIPTION
**Pull request by iAdramelk**
How to build:

```
yarn
yarn git-update
yarn build-node
yarn mintlify
```

After what use `migration-result` folder as a mintlify files.

Known limitations:

- The group title can't be a link in Mintlify, so I had to copy the title page inside the category, which lead to title duplication. See for example: `Cluster Access and RBAC / Cluster Access and RBAC`.
- Mintlify does not allow setting titles in navigation; it always takes the title from the page's frontmatter. It resulted in some long titles in the menu.
- Content from all versions of `ScopedBlock` is left in the code.
- Captions from `Figure` are removed.
- `Details` are always closed and info about min/max version is removed.

### CODE REVIEW

File: migration/plugins/remark-migrate-tags.mjs
Comment (avatus) [inline migration/plugins/remark-migrate-tags.mjs:17]: No need to make changes, but rather than a function you could just use a map of `suppliedName`->`desiredName` like this

```js
const names = {
    "tip": "Tip",
    "warning": "Warning",
}
```

and then access it using `names[suppliedName]`. I believe this would be more perform a o(1) lookup than a function call but in this case its completely negligible. Just felt like typing that out, no need to make any changes

**Review by avatus — APPROVED** [review]
Looks fine to me. Would love another set of eyes from @ptgott on the functionality!

File: migration/plugins/remark-migrate-tags.mjs
Comment (iAdramelk) [reply migration/plugins/remark-migrate-tags.mjs:17]: It will not work this way, unfortunately. Type is optional here, so we need to return `Note` not only for `type="note"`, but also for missing type.

Comment (avatus) [reply migration/plugins/remark-migrate-tags.mjs:17]: A 'names[suppliedName] || "Note"' would make it work with optional types. But either way, not a necessary change. Thanks for checking it out!

File: migration/index.mjs
Comment (ptgott) [inline migration/index.mjs:138]: I would indicate what processing entails here

**Review by ptgott — COMMENTED** [review]
Is there bandwidth to add unit tests? I think testing will make it a lot easier to:
- Understand the intended migration logic
- Make changes to fix papercuts as we get ready to make the migration permanent

I think running test cases against `processFile` would be a good place to start, since we can create vfiles in the test suite and make assertions against the resulting strings.

Test cases for `processEntry` would be good as well. Otherwise, it's a little difficult for me to tell what the intended behavior of the function is.

**Review by ptgott — COMMENTED** [review]
Thanks for the updates! I like that the migration plugins and their tests are treated the same way as the original `remark` plugins/tests. 

I haven't given the code a thorough look—since the site renders on the dev server and there's a test suite, I think it'll be fine to wait until we're observing the staging site to make adjustmenets.

@avatus Is there anything else we should do before we merge this?

**Review by avatus — APPROVED** [review]
Seems fine to me!

**Pull request merged**
